stl/inc/vector

```diff
@@ -2444,9 +2444,6 @@ private:
     using _Mycont          = typename _Mybase::_Mycont;
     using _Difference_type = typename _Mybase::_Difference_type;
 
-    // TRANSITION, ABI: non-trivial constructor

```

Comment:
No change requested: I found when this comment was introduced, by MSVC-PR-197682 "Annotate all the noexcept things" on 2019-08-19, which changed:

```cpp
_Vb_reference() noexcept { // construct with null pointer (private)
}
```

to:

```cpp
// TRANSITION, ABI: non-trivial constructor
_Vb_reference() = default;
```

This comment doesn't appear to be a "don't mess with this" comment (like `tuple<>`'s copy ctor, which *should* be defaulted but *must not* until we can break ABI, as I learned the hard way). As this was defaulted for 5 years, I believe we should be able to remove it.

Looking at the MSVC-PR's review history, originally `// TRANSITION, ABI` was being added to the original user-provided definition, because Billy thought that the Standard required `vector<bool>::reference` to be trivial (so the comment was saying "I know this makes the type non-trivial"). Then @CaseyCarter countered that the Standard didn't require that and also presciently observed that "If anything, the Standard wants this constructor to be deleted". At that point Billy updated the code to what was merged. At the time, `_Iterator_base12` had a user-provided default ctor, which was changed to defaulted with `nullptr` DMIs by #1546 on 2021-02-03. At that point, I believe the comment became completely inaccurate.

So I think this comment has been stale for a long time, and removing it is fine.

Reply:
> Then @CaseyCarter countered that the Standard didn't require that and also presciently observed that "If anything, the Standard wants this constructor to be deleted".

Regularity goes straight out the window for proxy reference types. Default construction is at best a footgun.

Thanks for the archaeology!